### PR TITLE
Undo orientation in WriteToPixelCallbackStage.

### PR DESCRIPTION
**Pull request by szabadka**

### CODE REVIEW

File: lib/jxl/dec_frame.h [hunk 1]
```diff
@@ -223,13 +223,14 @@ class FrameDecoder {
   // orientation. Performing this operation is not yet supported, so this
```
Comment (veluca93) [inline lib/jxl/dec_frame.h:223]: out of date :)

File: lib/jxl/render_pipeline/stage_write.cc [hunk 2]
```diff
@@ -334,17 +339,36 @@ class WriteToPixelCallbackStage : public RenderPipelineStage {
...
+        for (size_t i = 0; i < num; i += num_channels_) {
+          for (size_t c = 0; c < num_channels_; ++c) {
+            std::swap(temp[i + c], temp[last - i + c]);
+          }
+        }
+        xstart = width_ - xstart - xlen;
+      }
+      if (transpose_) {
+        for (size_t i = 0, j = 0; i < xlen; ++i, j += num_channels_) {
+          pixel_callback_.run(run_opaque_, thread_id, ypos, xstart + i, 1,
```
Comment (veluca93) [inline lib/jxl/render_pipeline/stage_write.cc:366]: While this works, it is *rather slow*. Maybe we should (eventually) think of faster ways to do it.

File: lib/jxl/dec_frame.h (hunk 1 above)
Comment (szabadka) [reply lib/jxl/dec_frame.h]: updated

File: lib/jxl/render_pipeline/stage_write.cc (hunk 2 above)
Comment (szabadka) [reply lib/jxl/render_pipeline/stage_write.cc:366]: There are some tricks to do NxN transpositions with SIMD, so maybe we could buffer some lines here, and output N pixels at each call. (But transposition is an inherently slow operation and the implementation we have currently does not seem to be much better.)

Comment (veluca93) [reply lib/jxl/render_pipeline/stage_write.cc:366]: Buffering lines wouldn't necessarily work unless we keep large buffers, it is probably better to buffer i.e. 8x8 chunks and guarantee in the RP that dimensions satisfy this-and-that constraint... Anyway, we don't need to do this now, just adding a TODO is fine.

Comment (szabadka) [reply lib/jxl/render_pipeline/stage_write.cc:367]: added the TODO

**Review by veluca93 — APPROVED** [review]

**Review by veluca93 — APPROVED** [review]

**Comment by eustas** [comment]
The following tests started failing:
```
The following tests FAILED:
	1869 - DecodeTest/DecodeTestParam.PixelTest/280x12RGBAtoRGBAf32BECallbackEarlyBufferO2  # GetParam() = 280x12RGBAtoRGBAf32BECallbackEarlyBufferO2 (Failed)
	2037 - DecodeTest/DecodeTestParam.PixelTest/280x12RGBAtoRGBAf32BECallbackEarlyBufferO3  # GetParam() = 280x12RGBAtoRGBAf32BECallbackEarlyBufferO3 (Failed)
	2709 - DecodeTest/DecodeTestParam.PixelTest/280x12RGBAtoRGBAf32BECallbackEarlyBufferO7  # GetParam() = 280x12RGBAtoRGBAf32BECallbackEarlyBufferO7 (Failed)
	2877 - DecodeTest/DecodeTestParam.PixelTest/280x12RGBAtoRGBAf32BECallbackEarlyBufferO8  # GetParam() = 280x12RGBAtoRGBAf32BECallbackEarlyBufferO8 (Failed)
```

**Pull request merged**